include/substrait/function/Function.h

```diff
@@ -95,7 +95,7 @@ struct FunctionImplementation {
   ParameterizedTypePtr returnType;
   std::optional<FunctionVariadic> variadic;
 
-  /// Test if the actual types matched with this function variant.
+  /// Test if the actual types matched with this function implement.
```

Comment:
It's longer but would this read better if it said "this function's implementation"?

Reply:
Rebased and updated